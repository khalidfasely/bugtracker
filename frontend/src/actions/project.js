import delBugApi from "../api/delBug";
import newBugApi from "../api/newBug";
import projectApi from "../api/project";

export const setProject = (project, bugs) => ({
    type: "SET_PROJECT",
    project,
    bugs
});

export const startSetProject = (pid) => {
    return (dispatch) => {
        return projectApi(pid).then((result) => {
            dispatch(setProject(result.project, result.bugs));
            return result;
        });
    };
};

export const setNewBug = (bug) => ({
    type: 'NEW_BUG',
    bug
});

export const startSetNewBug = (on_project, bugData) => {
    return (dispatch) => {
        return newBugApi(on_project, bugData).then(result => {
            if (result.bug) {
                dispatch(setNewBug(result.bug));
            }
            return result;
        });
    };
};

export const setDeleteBug = (bid) => ({
    type: 'DELETE_BUG',
    bid
});

export const startSetDeleteBug = (bid) => {
    return (dispatch) => {
        return delBugApi(bid).then(result => {
            dispatch(setDeleteBug(bid));
            return result;
        });
    };
};