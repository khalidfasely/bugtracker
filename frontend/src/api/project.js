export default async (pid) => {
    const res = await fetch(`/api/project/${pid}`);

    const project = await res.json();

    return project;
};