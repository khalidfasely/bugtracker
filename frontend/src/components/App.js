function App() {
  return (
    <div className="App">
      <header>
        Hello world from React        
      </header>
    </div>
  );
}

export default App;
