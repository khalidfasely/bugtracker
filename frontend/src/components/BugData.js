const BugData = ({ bug }) => (
    <div>
        <p>{bug?.id}</p>
        <p>{bug?.user?.username}</p>
        <p>{bug?.title}</p>
        <p>{bug?.description}</p>
        <p>{bug?.time}</p>
        <p>{bug?.classification}</p>
        <p>{`${bug?.active}`}</p>
        <div>
            {bug?.admins?.map(admin => <span key={admin} >{admin} </span>)}
        </div>
        <br />
        <div>
            {bug?.users_with?.map(user => <span key={user} >{user} </span>)}
        </div>
    </div>
);

export default BugData;