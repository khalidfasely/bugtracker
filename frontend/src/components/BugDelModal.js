import Modal from "react-modal";

const BugDelModal = ({ delModalOpen, setDelModalOpen, deleteBug }) => (
    <Modal
        isOpen={delModalOpen}
        onRequestClose={() => setDelModalOpen(false)}
    >
        If you delete this bug there is no way to return it!
        <button onClick={deleteBug} data-testid='delete_button'>Delete</button>
    </Modal>
);

export default BugDelModal;