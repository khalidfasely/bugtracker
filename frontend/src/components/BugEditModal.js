import Modal from "react-modal";

import NewBug from "./NewBug";

const BugEditModal = ({
    editModalOpen, setEditModalOpen,
    users, bug
}) => (
    <Modal
        isOpen={editModalOpen}
        onRequestClose={() => setEditModalOpen(false)}
    >
        <NewBug users={users} isEdit={true} bug={bug} setEditModalOpen={setEditModalOpen} />
    </Modal>
);

export default BugEditModal;