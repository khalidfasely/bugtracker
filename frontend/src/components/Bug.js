import { useState, useEffect } from "react";
import { connect } from "react-redux";
import { useParams } from "react-router-dom";
import { startSetBug } from "../actions/bug";

import BugItem from './BugItem';

export const Bug = ({ startSetBug }) => {
    const params = useParams();

    const [ableToSee, setAbleToSee] = useState(false);
    const [notFound, setNotFound] = useState('');

    useEffect(() => {
        startSetBug(params.pid, params.bid)
        .then((result) => {
            setNotFound('');

            if (result.message) {
                setNotFound(result.message);
            };
        })
        .then(() => setAbleToSee(true))
        .catch(er => console.error(er));
    }, [params])


    if (notFound) {
        return <div>Bug Not Found! -- {notFound}</div>;
    };

    return (
        <div>
            {
                ableToSee ?
                <div><BugItem /></div> :
                <div>You're not able to see this bug</div>
            }
        </div>
    );
};

const mapDispatchToProps = (dispatch) => ({
    startSetBug: (pid, bid) => dispatch(startSetBug(pid, bid))
});

export default connect(undefined, mapDispatchToProps)(Bug);