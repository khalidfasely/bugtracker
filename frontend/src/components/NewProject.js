import { useEffect, useState } from "react";
import { connect } from "react-redux";
import Select from 'react-select';
import makeAnimated from 'react-select/animated';
import { useNavigate } from 'react-router-dom';

import { startSetNewProject } from "../actions/newProject";
import { startSetUsers } from "../actions/users";

const NewProject = ({ uname, users, startSetNewProject, startSetUsers }) => {
    const history = useNavigate();

    const [projectName, setProjectName] = useState('');
    const [errorName, setErrorName] = useState('');
    const [selectedUsers, setSelectedUsers] = useState([{ value: uname, label: uname, isFixed: true}]);
    const [selectedAdmins, setSelectedAdmins] = useState([{ value: uname, label: uname, isFixed: true}]);
    const [errorSelect, setErrorSelect] = useState('');

    useEffect(() => {
        startSetUsers();
    }, [])

    let options = users?.map(user => ({ value: user, label: user }));


    const customTheme = (theme) => ({
        ...theme,
        colors: {
            ...theme.colors,
            primary25: 'orange',
            primary: 'red'
        }
    });

    const onChangeUsers = (selectedUsers, { action, removedValue }) => {
        switch (action) {
            case "remove-value":
            case "pop-value":
                if (removedValue.isFixed) {
                  return;
                }
                break;
            case "clear":
                selectedUsers = selectedUsers.filter(v => v.isFixed);
                break;
            default:
                break;
        };
        setSelectedUsers(selectedUsers);
    };

    const onChangeAdmins = (selectedAdmins, { action, removedValue }) => {
        switch (action) {
            case "remove-value":
            case "pop-value":
                if (removedValue.isFixed) {
                  return;
                }
                break;
            case "clear":
                selectedAdmins = selectedAdmins.filter(v => v.isFixed);
                break;
            default:
                break;
        };
        setSelectedAdmins(selectedAdmins);
    };

    const incorrectUsers = (users, admins) => {
        let arrUsers = users.map(user => user.value);
        let arrAdmins = admins.map(admin => admin.value);

        for (let i = 0; i < arrAdmins.length; i++) {
            if (!arrUsers.includes(arrAdmins[i])) {
                return 'Must all admins be users first!';
            }
        }
    };

    const tryCreateProject = (e) => {
        e.preventDefault();

        setErrorName('');
        setErrorSelect('');

        const isIncorrectUsers = incorrectUsers(selectedUsers, selectedAdmins);
        //correctUsers(selectedUsers, selectedAdmins);

        if (isIncorrectUsers) {
            setErrorSelect(isIncorrectUsers);
            return;
        }

        const correctProjectName = projectName.replace(/\s/g, '').length <= 15;

        if (correctProjectName) {
            setErrorName('Project\'s name must be bigger than 15 characters!');
        } else {
            let arrUsers = selectedUsers.map(user => user.value);
            let arrAdmins = selectedAdmins.map(admin => admin.value);
            startSetNewProject({ projectName, arrUsers, arrAdmins}).then(result => history(`/project/${result.project.id}`));
        };
    };

    return (
        <div>
            <form onSubmit={tryCreateProject}>
                {errorName && <p>{errorName}</p>}
                <input
                    name="name"
                    placeholder="Project's Name"
                    value={projectName}
                    onChange={(e) => setProjectName(e.target.value)}
                    autoFocus
                />
                {errorSelect && <p>{errorSelect}</p>}
                <Select
                    components={makeAnimated()}
                    value={selectedUsers}
                    theme={customTheme}
                    defaultValue={selectedUsers}
                    onChange={onChangeUsers}
                    options={options}
                    isClearable={!options && options.some(option => option.isFixed)}
                    placeholder='Select Users'
                    noOptionsMessage={() => 'No other user!'}
                    isMulti
                    isSearchable
                />
                <Select
                    components={makeAnimated()}
                    value={selectedAdmins}
                    theme={customTheme}
                    defaultValue={selectedAdmins}
                    onChange={onChangeAdmins}
                    options={selectedUsers}
                    isClearable={!options && options.some(option => option.isFixed)}
                    placeholder='Select Admins'
                    noOptionsMessage={() => 'No other user!'}
                    isMulti
                    isSearchable
                />
                <button>Create.</button>
            </form>
        </div>
    );
};

const mapStateToProps = (state) => ({
    uname: state.auth.uname,
    users: state.users
});

const mapDispatchToProps = (dispatch) => ({
    startSetNewProject: (project) => dispatch(startSetNewProject(project)),
    startSetUsers: () => dispatch(startSetUsers())
});

export default connect(mapStateToProps, mapDispatchToProps)(NewProject);