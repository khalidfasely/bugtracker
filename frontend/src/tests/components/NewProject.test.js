import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { NewProject } from '../../components/NewProject';
import project from '../fixtures/project';

let newProjectProps, editProjectProps;

beforeEach(() => {
    newProjectProps = {
        uname: 'admin',
        users: ['admin', '1'],
        startSetNewProject: jest.fn(() => Promise.resolve({ project: { id: 1 } })),
        startSetUsers: jest.fn()
    }
    editProjectProps = {
        isEdit: true,
        projectItemEdit: project,
        setEditModalOpen: jest.fn(),
        startSetEditProject: jest.fn(() => Promise.resolve({}))
    }
});

test('Should render NewProject component correctly without props(uname)', () => {
    const { asFragment } = render(<NewProject startSetUsers={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render NewProject component correctly with new project props', () => {
    const { asFragment } = render(<NewProject {...newProjectProps} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render NewProject component correctly with edit project props', () => {
    const { asFragment } = render(<NewProject {...{...newProjectProps, ...editProjectProps}} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should change name input correctly in new project', () => {
    const { getByTestId } = render(<NewProject {...newProjectProps} />);
    const nameInputEl = getByTestId('name_input');

    expect(nameInputEl.value).toBe('');

    fireEvent.change(nameInputEl, {
        target: { value: 'New name for the new project!' }
    });

    expect(nameInputEl.value).toBe('New name for the new project!');
});

test('Should change name input correctly in edit project', () => {
    const { getByTestId } = render(<NewProject {...{...newProjectProps, ...editProjectProps}} />);
    const nameInputEl = getByTestId('name_input');

    expect(nameInputEl.value).toBe(editProjectProps.projectItemEdit.name);

    fireEvent.change(nameInputEl, {
        target: { value: 'New name for the new project!' }
    });

    expect(nameInputEl.value).toBe('New name for the new project!');
});

test('Should display an error if name\'s length < 15', () => {
    const { getByTestId } = render(<NewProject {...newProjectProps} />);
    const formEl = getByTestId('newproject_form');
    const nameInputEl = getByTestId('name_input');
    const nameErrorEl = getByTestId('name_error');

    expect(nameErrorEl.textContent).toBe('');

    fireEvent.submit(formEl, {
        preventDefault: () => {}
    });

    expect(nameErrorEl.textContent).toBe('Project\'s name must be bigger than 15 characters!');

    fireEvent.change(nameInputEl, {
        target: { value: 'This is new project to make the error gone' }
    });

    fireEvent.submit(formEl, {
        preventDefault: () => {}
    });

    expect(nameErrorEl.textContent).toBe('');
});

test('Should fire startSetNewProject correctly', () => {
    const { getByTestId } = render(<NewProject {...newProjectProps} />);
    const formEl = getByTestId('newproject_form');
    const nameInputEl = getByTestId('name_input');

    fireEvent.change(nameInputEl, {
        target: { value: 'This project name is for test purpose' }
    });

    expect(newProjectProps.startSetNewProject).toHaveBeenCalledTimes(0);

    fireEvent.submit(formEl, {
        preventDefault: () => {}
    });

    expect(newProjectProps.startSetNewProject).toHaveBeenCalledTimes(1);
});

test('Should fire startSetEditProject correctly', () => {
    const { getByTestId } = render(<NewProject {...{...newProjectProps, ...editProjectProps}} />);
    const formEl = getByTestId('newproject_form');
    const nameInputEl = getByTestId('name_input');

    fireEvent.change(nameInputEl, {
        target: { value: 'This project name is for test purpose' }
    });

    expect(editProjectProps.startSetEditProject).toHaveBeenCalledTimes(0);

    fireEvent.submit(formEl, {
        preventDefault: () => {}
    });

    expect(editProjectProps.startSetEditProject).toHaveBeenCalledTimes(1);
});