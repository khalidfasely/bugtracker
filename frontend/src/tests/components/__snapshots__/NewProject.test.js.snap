// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NewProject component correctly with edit project props 1`] = `
<DocumentFragment>
  <div
    class="project-form-container"
  >
    <form
      data-testid="newproject_form"
    >
      <p
        data-testid="name_error"
      />
      <input
        class="project-form__input"
        data-testid="name_input"
        maxlength="120"
        name="name"
        placeholder="Project's Name"
        value="Project one"
      />
      <div>
        <div
          class="select-users css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-4-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1s2u09g-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
              style="display: flex;"
            >
              <div
                style="overflow: hidden; white-space: nowrap; width: auto;"
              >
                <div
                  class="css-1rhbuit-multiValue"
                >
                  <div
                    class="css-12jo7m5"
                  >
                    1
                  </div>
                  <div
                    aria-label="Remove 1"
                    class="css-xb97g8"
                    role="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="14"
                      viewBox="0 0 20 20"
                      width="14"
                    >
                      <path
                        d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                style="overflow: hidden; white-space: nowrap; width: auto;"
              >
                <div
                  class="css-1rhbuit-multiValue"
                >
                  <div
                    class="css-12jo7m5"
                  >
                    admin
                  </div>
                  <div
                    aria-label="Remove admin"
                    class="css-xb97g8"
                    role="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="14"
                      viewBox="0 0 20 20"
                      width="14"
                    >
                      <path
                        d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-4-listbox"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-owns="react-select-4-listbox"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-4-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="select-users css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-5-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1s2u09g-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
              style="display: flex;"
            >
              <div
                style="overflow: hidden; white-space: nowrap; width: auto;"
              >
                <div
                  class="css-1rhbuit-multiValue"
                >
                  <div
                    class="css-12jo7m5"
                  >
                    1
                  </div>
                  <div
                    aria-label="Remove 1"
                    class="css-xb97g8"
                    role="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="14"
                      viewBox="0 0 20 20"
                      width="14"
                    >
                      <path
                        d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-5-listbox"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-owns="react-select-5-listbox"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-5-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="buttons-container"
      >
        <button>
          Save
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Should render NewProject component correctly with new project props 1`] = `
<DocumentFragment>
  <div
    class="project-form-container"
  >
    <form
      data-testid="newproject_form"
    >
      <p
        data-testid="name_error"
      />
      <input
        class="project-form__input"
        data-testid="name_input"
        maxlength="120"
        name="name"
        placeholder="Project's Name"
        value=""
      />
      <div>
        <div
          class="select-users css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1s2u09g-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
              style="display: flex;"
            >
              <div
                style="overflow: hidden; white-space: nowrap; width: auto;"
              >
                <div
                  class="css-1rhbuit-multiValue"
                >
                  <div
                    class="css-12jo7m5"
                  >
                    admin
                  </div>
                  <div
                    aria-label="Remove admin"
                    class="css-xb97g8"
                    role="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="14"
                      viewBox="0 0 20 20"
                      width="14"
                    >
                      <path
                        d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-2-listbox"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-owns="react-select-2-listbox"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-2-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="select-users css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1s2u09g-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
              style="display: flex;"
            >
              <div
                style="overflow: hidden; white-space: nowrap; width: auto;"
              >
                <div
                  class="css-1rhbuit-multiValue"
                >
                  <div
                    class="css-12jo7m5"
                  >
                    admin
                  </div>
                  <div
                    aria-label="Remove admin"
                    class="css-xb97g8"
                    role="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="14"
                      viewBox="0 0 20 20"
                      width="14"
                    >
                      <path
                        d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-3-listbox"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-owns="react-select-3-listbox"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-3-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="buttons-container"
      >
        <button>
          Create
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Should render NewProject component correctly without props(uname) 1`] = `
<DocumentFragment>
  <div>
    <a>
      Login
    </a>
     or 
    <a>
      Sign In
    </a>
     to Add a Project!
  </div>
</DocumentFragment>
`;
